crates/red_knot_python_semantic/src/types.rs

```diff
@@ -306,3 +312,359 @@ pub struct IntersectionType<'db> {
     // the intersection type does not include any value in any of these types
     negative: FxOrderSet<Type<'db>>,
 }
+
+#[allow(unused)]
+#[derive(Clone)]
+struct IntersectionTypeBuilder<'db> {
```

Comment:
nit: Does `Type` add anything here? `IntersectionBuilder`/`IntersectionBuilderInner` might be slightly shorter than `IntersectionTypeBuilder`/`InnerIntersectionTypeBuilder`.

Reply:
Yep, agreed, I'll remove `Type` from both of these names.